### PR TITLE
common platform

### PR DESCRIPTION
**Pull request by adecaro**
This PR does the following: It introduces a `common` platform that contains shared code between the fabric and the orion platform. This code can be of help for other platforms in the future.
In addition, this PR removes `TxStatusChangeListener` and replace it `FinalityListener`.

### CODE REVIEW

File: platform/fabric/core/generic/committer/em.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	c.EventQueue <- event
+}
+
+func (c *EventManager) Dispatch(event TxEvent) {
+	defer func() {
+		if r := recover(); r != nil {
+			logger.Errorf("caught panic while running dispatching event [%v]: [%s][%s]", event, r, debug.Stack())
+		}
+	}()
+	l := c.cloneListeners(event.TxID)
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/committer/em.go:91]: `l` can be nil

File: platform/fabric/core/generic/committer/em.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	for _, listener := range l {
+		if err := listener.OnStatus(event.TxID, int(event.ValidationCode), event.ValidationMessage); err != nil {
+			logger.Errorf("failed on status call [%v]: [%s][%s]", event, err, debug.Stack())
+		}
+		c.DeleteListener(event.TxID, listener)
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/committer/em.go:96]: You could delete all listeners the moment you clone them, right? We don't have any revert mechanism and we keep the write lock for O(n^2) time like this while we can do it in O(1).

File: platform/fabric/core/generic/committer/em.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	for {
+		select {
+		case <-context.Done():
+			return
+		case <-ticker.C:
+			txIDs := c.txIDs()
+			for _, txID := range txIDs {
+				// check txID status, if it is valid or invalid, post an event
+				logger.Debugf("check tx [%s]'s status", txID)
+				vc, message, err := c.Vault.Status(txID)
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/committer/em.go:128]: We could have a `Status` method that accepts multiple txIDs to optimise the query

File: platform/fabric/core/generic/committer/em.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	} else {
+		var ok bool
+		ls, ok = c.txIDListeners[txID]
+		if !ok {
+			return
+		}
+	}
+	for i, l := range ls {
+		if l == ch {
+			ls = append(ls[:i], ls[i+1:]...)
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/committer/em.go:74]: We need to delete the key if len(ls) == 0. Otherwise it stays in the txIDs() result.

**Review by alexanderfilios — APPROVED** [review]
LGTM

**Pull request merged**